src/gpasswd.c

```diff
@@ -818,7 +818,7 @@ static void change_passwd (struct group *gr)
 #endif
 {
 	char *cp;
-	static char pass[BUFSIZ];
+	static char pass[PASS_MAX + 1];
```

Comment:
On the other hand, I'm thinking that there's a further improvement here:

We could define a `passdup()` function which fuplicates a function via strdup(3).

The benefit over an array is that the size is always correct.
The benefit over raw strdup(3) is that we can mark it with an attribute so that it must be destroyed via erase_pass().

I'll prepare a patch for after we merge yours.